# Transcriptional Regulatory Element Database

<https://cb.utdallas.edu/cgi-bin/TRED/>

# Introduction

In order to understand gene regulation, accurate and comprehensive knowledge
of transcriptional regulatory elements is essential. Transcriptional
Regulatory Element Database (TRED) has been built in response to increasing
needs of an integrated repository for both cis- and trans- regulatory elements
in mammals, and the lack of such resources at present. 

Genome-wide human, mouse and rat promoter annotation in TRED was realized by
an automated pipeline to extract known promoters from databases such as
Genbank, EPD and DBTSS, and employ promoter finding program FirstEF combined
with mRNA/EST information and cross-species comparisons. We have also carried
out hand curation to assess computational prediction and ensure data accuracy.
A quality level is assigned to each promoter based on the reliability of the
supporting evidence. 

Curation has also been done for transcriptional regulation information,
including transcription factor binding motifs and experimental evidence.
Binding motifs are mapped on promoters of the corresponding genes and binding
quality levels are assigned based on definitiveness of the binding evidence.
Curation is currently focusing on target genes of 36 cancer-related TF
families. 

Distinguishing features of TRED include:

- relatively complete genome wide promoter annotation for human, mouse and rat
- availability of transcription factor binding and regulation information
- data accuracy is ensured by curation, which continues to expand
- easy and flexible data retrieval
- availability of on-the-fly sequence analysis tools


TRED can provide good training datasets for further genome wide cis-regulatory
element prediction, assist detailed functional studies, and facilitate to
decipher the gene regulatory networks. 

Michael Zhang Lab, Cold Spring Harbor Laboratory, all rights reserved
Questions/suggestions email: Ashwinikumar Kulkarni <aak093020@utdallas.edu>
